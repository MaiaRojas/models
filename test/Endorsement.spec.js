const mongoose = require('mongoose');
const { Endorsement } = require('../')(mongoose);

describe('Endorsement', () => {
  beforeAll(async () => {
    await mongoose.connect(global.__MONGO_URI__, { useNewUrlParser: true });
  });

  afterAll(async () => {
    await mongoose.disconnect();
  });

  beforeEach(async () => {
    await mongoose.connection.db.dropDatabase();
    await Endorsement.createIndexes();
  });

  it('should fail when missing props', () => {
    const endorsement = new Endorsement();
    return endorsement.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should fail when endorsement is not a string', () => {
    const endorsement = new Endorsement({
      text: [],
    });
    return endorsement.save()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should save successfully with appropriate fields', () => {
    const endorsement = new Endorsement({
      text: 'foo',
    });

    return endorsement.save()
      .then((result) => {
        expect(typeof result.text).toBe('string');
        expect(result.text).toBe(endorsement.text);
      });
  });
});
