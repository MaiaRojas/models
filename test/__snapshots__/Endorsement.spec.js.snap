// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Endorsement should fail when endorsement is not a string 1`] = `
Object {
  "text": [CastError: Cast to String failed for value "[]" at path "text"],
}
`;

exports[`Endorsement should fail when missing props 1`] = `
Object {
  "text": [ValidatorError: Path \`text\` is required.],
}
`;
